### PR TITLE
Adjust connection windows size dynamically according to ACK received

### PR DESCRIPTION
**Pull request by billfort**
Signed-off-by: bill fort <fxbao@hotmail.com>

### CODE REVIEW

File: src/session.js [hunk 1]
```diff
@@ -390,16 +394,17 @@ export default class Session {
...
-      let conn = new Connection(this, this.localClientIDs[i], packet.getClientIdsList()[i]);
+      let conn = new Connection(this, this.localClientIDs[i], packet.getClientIdsList()[i], initialWindowSize);
       connections.set(util.connKey(conn.localClientID, conn.remoteClientID), conn);
     }
     this.connections = connections;
 
     this.remoteClientIDs = packet.getClientIdsList();
     this.sendChan = new Channel();
-    this.resendChan = new Channel(this.config.maxConnectionWindowSize * n);
+    this.resendChan = new Channel(this.sendWindowPacketCount);
```
Comment (yilunzhang) [inline src/session.js]: +n

Comment (billfort) [reply src/session.js]: Yes, modified

File: src/session.js [hunk 2]
```diff
@@ -77,6 +78,7 @@ export default class Session {
     this._readableStream = null;
     this.WritableStream = null;
     this._writableStream = null;
+    this.sendWindowPacketCount=this.sendWindowSize / this.sendMtu;
```
Comment (yilunzhang) [inline src/session.js]: space

Comment (billfort) [reply src/session.js]: Yes, modified.

**Review by yilunzhang — APPROVED** [review]
Looks good to me

**Pull request merged**